grc/core/blocks/embedded_python.py

```diff
@@ -149,10 +149,10 @@ def rewrite(self):
     def _update_params(self, params_in_src):
         param_factory = self.parent_platform.make_param
         params = {}
-        for param in list(self.params):
-            if hasattr(param, '__epy_param__'):
-                params[param.key] = param
-                del self.params[param.key]
+        for key, value in self.params.copy().items():
```

Comment:
An open Q was, does this work in Python2.

Reply:
oh, that's the reason I was fighting with oldish containers to even get to try that.

Reply:
I forgot what the reason was and gave up. Now I have new motivation. Also, we need internal unit test for GUI stuff, honestly, at some point.

Reply:
(by the way, yes, we should be testing things with Py2. Why are you esp. nervous about this line?)

Reply:
I can't remember...

Reply:
This works in Py2k.